File: android/src/org/colinw/asteroids/BBox.java
package org.colinw.asteroids;

public class BBox {
  private boolean set_;
  private double  xmin_, ymin_, xmax_, ymax_;
  
  BBox() {
    set_ = false;
  }
  
  BBox(double xmin, double ymin, double xmax, double ymax) {
    xmin_ = xmin;
    ymin_ = ymin;
    xmax_ = xmax;
    ymax_ = ymax;
    
    set_ = true;
  }
  
  public double xmin() { return xmin_; }
  public double ymin() { return ymin_; }
  public double xmax() { return xmax_; }
  public double ymax() { return ymax_; }
  
  void reset() {
    set_ = false;
  }
  
  void add(double x, double y) {
    if (! set_) {
      xmin_ = x; ymin_ = y;
      xmax_ = x; ymax_ = y;
      
      set_ = true;
    }
    else {
      xmin_ = Math.min(xmin_, x);
      ymin_ = Math.min(ymin_, y);
      xmax_ = Math.max(xmax_, x);
      ymax_ = Math.max(ymax_, y);
    }
  }
  
  boolean inside(double x, double y) {
    return ((x >= xmin_ && x <= xmax_) && (y >= ymin_ && y <= ymax_));
  }
}

File: html/bbox.js
'use strict';

//------

function BBox (x1, y1, x2, y2) {
  if (x1 != undefined) {
    this.x1 = x1;
    this.y1 = y1;
    this.x2 = x2;
    this.y2 = y2;

    this.set = true;
  }
  else {
    this.set = false;
  }
}

BBox.prototype.reset = function() {
  this.set = false;
};

BBox.prototype.add = function(p) {
  if (! this.set) {
    this.x1 = p.x;
    this.y1 = p.y;
    this.x2 = p.x;
    this.y2 = p.y;

    this.set = true;
  }
  else {
    this.x1 = Math.min(this.x1, p.x);
    this.y1 = Math.min(this.y1, p.y);
    this.x2 = Math.max(this.x2, p.x);
    this.y2 = Math.max(this.y2, p.y);
  }
};

BBox.prototype.inside = function(p) {
  if (! this.set)
    return false;

  return (p.x >= this.x1 && p.x <= this.x2 && p.y >= this.y1 && p.y <= this.y2);
};
